Ensure default behaviour when (suite case) id is 0 (#177)
eclipse/kiso-testing: docs/getting_started/basic_tests.rst

- ``case_id``: current test case identification number (optional for test suite setup and teardown)
- ``aux_list``: list of used auxiliaries

> ``suite_id`` and ``case_id`` are used to clearly define a test execution order but remain optional(still remain
> mandatory for remote testing). If both are not defined (by default ``suite_id`` and ``case_id`` are equal to 0),

Reviewer: ```suggestion
``suite_id`` and ``case_id`` are used to coordinate and define a clear execution order.  It is now optional for ``pykiso.BasicTest`` but still mandatory in case of ``pykiso.RemoteTest``.
If both ids  are not defined (by default ``suite_id`` and ``case_id`` are equal to 0),
```

the alphabetical order will be applied on each .py modudle and each contained test class.

In other words, If we have the following test suite folder organisation:

---> test_suite_folder
 |
 ----> a.py module with classes B/C/A (declare in this order)
 |
 ----> b.py module with classes Z/G (declare in this order)
 |
 ----> c.py module with classes S/T + Suite Setup/Teardown

The framework will execute the tests in the below order:

 .. code:: bash

    test_suite_setUp (c.SuiteSetup-0-0)
    test_run (a.A-0-0)
    test_run (a.B-0-0)

>     test_run (a.C-0-0)

Reviewer: It actually would be cool, if none are defined, we do not print the -0-0? no?

Author: This representation is the print of the internal test suite object who collect the tests.

- ``case_id``: current test case identification number (optional for test suite setup and teardown)
- ``aux_list``: list of used auxiliaries

``suite_id`` and ``case_id`` are used to coordinate and define a clear test execution order.

> It is now optional for ``pykiso.BasicTest`` but still mandatory in case of ``pykiso.RemoteTest``.

Reviewer 2: ```suggestion
It is now optional for ``pykiso.BasicTest`` but still mandatory in case of ``pykiso.RemoteTest`` as the TestApp protocol relies on these identifiers.
```
(maybe with a ref to the testapp protocol)

> If both ids are not defined (by default ``suite_id`` and ``case_id`` are equal to 0),

Reviewer 2: ```suggestion
If none of these IDs are defined, the default value of 0 will be used and
```

> the alphabetical order will be applied on each .py modudle and each contained test class.

Reviewer 2: ```suggestion
the alphabetical order will be applied on each .py module and each contained test class.
```

In other words, If we have the following test suite folder organisation:

---> test_suite_folder
 |
 ----> a.py module with classes B/C/A (declare in this order)
 |
 ----> b.py module with classes Z/G (declare in this order)
 |
 ----> c.py module with classes S/T + Suite Setup/Teardown

> The framework will execute the tests in the below order:

Reviewer 2: ```suggestion
The framework will execute the tests in the following order:
```

 .. code:: bash

    test_suite_setUp (c.SuiteSetup-0-0)
    test_run (a.A-0-0)
    test_run (a.B-0-0)
    test_run (a.C-0-0)
    test_run (b.G-0-0)
    test_run (b.Z-0-0)
    test_run (c.S-0-0)
    test_run (c.T-0-0)
    test_suite_tearDown (c.SuiteTearDown-0-0)

> In order to utilise the SetUp/TearDown test-suite feature, users have to define a class inheriting from

Reviewer 2: ```suggestion
In order to make use of the SetUp/TearDown test-suite feature, users have to define a class inheriting from
```
